packages/esm-patient-search-app/src/patient-search-page/refine-search/person-attributes.resource.ts

```diff
@@ -0,0 +1,105 @@
+import useSWRImmutable from 'swr/immutable';
+import {
+  type FetchResponse,
+  fhirBaseUrl,
+  openmrsFetch,
+  restBaseUrl,
+  showSnackbar,
+  useDebounce,
+} from '@openmrs/esm-framework';
+import { useMemo } from 'react';
+import useSWR from 'swr';
+import {
+  type ConceptAnswers,
+  type ConceptResponse,
+  type LocationEntry,
+  type LocationResponse,
+  type PersonAttributeTypeResponse,
+} from '../../types';
+
+export function useAttributeConceptAnswers(conceptUuid: string): {
+  data: Array<ConceptAnswers>;
+  isLoading: boolean;
+  error: Error | undefined;
+} {
+  const shouldFetch = typeof conceptUuid === 'string' && conceptUuid !== '';
+  const { data, error, isLoading } = useSWRImmutable<FetchResponse<ConceptResponse>, Error>(
+    shouldFetch ? `${restBaseUrl}/concept/${conceptUuid}` : null,
+    openmrsFetch,
+  );
+  if (error) {
+    showSnackbar({
+      title: error.name,
+      subtitle: error.message,
+      kind: 'error',
+    });
+  }
```

Comment:
* This should be wrapped inside an useEffect, else it will show snackbar for every rerender of the respective component.
* For errors like these, we show Toast instead of a Snackbar.

Reply:
> For errors like these, we show Toast instead of a Snackbar.

Actually, its better to show errors where the user can't do anything about it anyways via a Snackbar. Toasts should only be used for critical situations, i.e., where the user needs to take an action. Anything purely informative—snackbar.